core/server/api_container/server/startosis_engine/kurtosis_instruction/get_services/get_services.go

```diff
@@ -0,0 +1,103 @@
+package get_services
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/enclave_plan_persistence"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/enclave_structure"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/interpretation_time_value_store"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_starlark_framework"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/builtin_argument"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_plan_instruction"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/plan_yaml"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/startosis_errors"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/startosis_validator"
+	"go.starlark.net/starlark"
+)
+
+const (
+	GetServicesBuiltinName = "get_services"
+	descriptionStr         = "Fetching services"
+)
+
+func NewGetServices(interpretationTimeStore *interpretation_time_value_store.InterpretationTimeValueStore) *kurtosis_plan_instruction.KurtosisPlanInstruction {
+	return &kurtosis_plan_instruction.KurtosisPlanInstruction{
+		KurtosisBaseBuiltin: &kurtosis_starlark_framework.KurtosisBaseBuiltin{
+			Name:        GetServicesBuiltinName,
+			Arguments:   []*builtin_argument.BuiltinArgument{},
+			Deprecation: nil,
+		},
+		Capabilities: func() kurtosis_plan_instruction.KurtosisPlanInstructionCapabilities {
+			return &GetServicesCapabilities{
+				interpretationTimeStore: interpretationTimeStore,
+				serviceNames:            []service.ServiceName{}, // populated at interpretation time
+				description:             "",                      // populated at interpretation time
+			}
+		},
+		DefaultDisplayArguments: map[string]bool{},
+	}
+}
+
+type GetServicesCapabilities struct {
+	interpretationTimeStore *interpretation_time_value_store.InterpretationTimeValueStore
+	serviceNames            []service.ServiceName
+	description             string
+}
+
+func (builtin *GetServicesCapabilities) Interpret(_ string, arguments *builtin_argument.ArgumentValuesSet) (starlark.Value, *startosis_errors.InterpretationError) {
+	builtin.description = builtin_argument.GetDescriptionOrFallBack(arguments, descriptionStr)
+
+	// Right now, due to semantics of interpretationTimeStore this will return all services that have been added during interpretation, not accounting for services removed
+	// TODO: Remove service from interpretationTimeStore during plan.remove_service()
```

Comment:
maybe i should actually just impl this now on second thought I think it should be expected behavior

Reply:
i'd expect it to be running services; can we have a look at the ticket again?

how would i as the end user know what are the running services based on this list? I don't want to make exec calls to a dead service & fail

Reply:
perhaps our existing validation logic will catch the dead service being exec'd on? how do we handle

```
add A
remove A
add A
```

which A do we get?

Reply:
in this scenario our existing logic `get_service` and a `get_services` call would give you get the second A, as expected.